### PR TITLE
Add support for "isOpen" prop on popup

### PR DESCRIPTION
**Pull request by nbramblett**
When this prop is defined (either true or false), it overrides the internal "open" controls and defers that control entirely to the parent component.

### CODE REVIEW

File: test-site/src/App.tsx [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import { ChatHeader, ChatPanel, ChatPopUp } from "@yext/chat-ui-react";
+import { ChatPopUpProps } from "../../lib/esm/src/components/ChatPopUp";
```
Comment (yen-tt) [inline test-site/src/App.tsx:2]: nit: lets import this in line 1 rather than through lib

File: src/components/ChatPopUp.tsx [hunk 2]
```diff
@@ -179,6 +168,16 @@ export function ChatPopUp(props: ChatPopUpProps) {
   const isOpenOnLoad =
     (messages.length > 1 && openOnLoadLocalStorage === "true") || openOnLoad;
 
+  // Set the initial value of the local storage flag for opening on load only if it doesn't already exist
```
Comment (yen-tt) [inline src/components/ChatPopUp.tsx:171]: this comment is meant for L157 no?

File: src/components/ChatPopUp.tsx [hunk 3]
```diff
@@ -298,3 +287,62 @@ export function ChatPopUp(props: ChatPopUpProps) {
...
+  const [showInitialMessage, setshowInitialMessage] = useState(
+    initialMessageVisible
+  );
+
+  useEffect(() => {
+    if (isOpen !== undefined) {
+      setShowChat(isOpen);
+      setRenderChat(isOpen);
+    }
+  }, [renderChat, isOpen]);
```
Comment (yen-tt) [inline src/components/ChatPopUp.tsx:310]: why is `renderChat` needed in deps list?

Comment (nbramblett) [reply src/components/ChatPopUp.tsx:310]: oh that's weird, eslint didn't catch this. fixed

**Pull request merged**